src/fsharp/ErrorResolutionHints.fs

```diff
@@ -9,26 +9,28 @@ let highConfidenceThreshold = 0.85
 let minStringLengthForThreshold = 3
 
 /// We report a candidate if its edit distance is <= the threshold.
-/// The threshhold is set to about a quarter of the number of characters the user entered.
-let IsInEditDistanceProximity userEntered suggestion =
-    let editDistance = Internal.Utilities.EditDistance.CalcEditDistance(userEntered,suggestion)
+/// The threshhold is set to about a quarter of the number of characters.
+let IsInEditDistanceProximity idText suggestion =
+    let editDistance = Internal.Utilities.EditDistance.CalcEditDistance(idText,suggestion)
     let threshold =
-        match userEntered.Length with
+        match idText.Length with
         | x when x < 5 -> 1
         | x when x < 7 -> 2
         | x -> x / 4 + 1
     
     editDistance <= threshold
 
 /// Filters predictions based on edit distance to the given unknown identifier.
-let FilterPredictions (userEntered:string) (suggestionF:ErrorLogger.Suggestions) =    
-    let uppercaseText = userEntered.ToUpperInvariant()
-    suggestionF()
+let FilterPredictions (idText:string) (suggestionF:ErrorLogger.Suggestions) =    
+    let uppercaseText = idText.ToUpperInvariant()
+    let allSuggestions = suggestionF()
+
+    if allSuggestions.Contains idText then [] else // some other parsing error occurred
```

Comment:
Here you use contains idText a little later you use endswith '.idText'.
Which is the correct test? 'idText' or '"." + idText'

Does suggestion return UpperInvariant()?

if not then if allSuggestions.Contains idText test is probably wrong.
otherwise perhaps:
let suggestedText = suggestion.ToUpperInvariant() 

is redundant.